### PR TITLE
Link check should be initially passing if min links is zero. Fixes #735

### LINKED ISSUE #735: router link health check should only be passing initially if min links is zero

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: router/config.go [hunk 1]
```diff
@@ -664,22 +666,30 @@ func LoadConfig(path string) (*Config, error) {
...
 						if val, ok := value.(int); ok {
 							cfg.HealthChecks.LinkCheck.MinLinks = val
 						} else {
 							return nil, errors.Wrapf(err, "invalid value [%v] for healthChecks.linkCheck.minLinks", value)
 						}
 					}
 
+					if value, found := checkMap["initialDelay"]; found {
+						if val, err := time.ParseDuration(fmt.Sprintf("%v", value)); err == nil {
+							cfg.HealthChecks.LinkCheck.InitialDelay = val
```
Comment (dovholuknf) [inline router/config.go:687]: accompanying doc/examples are updated elsewhere i take it, right?

File: router/router.go [hunk 2]
```diff
@@ -586,8 +586,8 @@ func (self *Router) initializeHealthChecks() (gosundheit.Health, error) {
 	err = h.RegisterCheck(&linkHealthCheck{router: self, minLinks: checkConfig.LinkCheck.MinLinks},
 		gosundheit.ExecutionPeriod(checkConfig.LinkCheck.Interval),
 		gosundheit.ExecutionTimeout(5*time.Second),
-		gosundheit.InitiallyPassing(true),
-		gosundheit.InitialDelay(1*time.Second),
+		gosundheit.InitiallyPassing(checkConfig.LinkCheck.MinLinks == 0),
```
Comment (dovholuknf) [inline router/router.go:589]: a thought... feels like logic that should be in a separate function that's eventually testable/assert-able?

**Review by dovholuknf — APPROVED** [review]
looks ok to me...

File: router/router.go (hunk 2 above)
Comment (plorenz) [reply router/router.go:589]: Yes. I would wait to refactor it until we add the test, other wise it'll likely need to be refactored twice :)

File: router/config.go (hunk 1 above)
Comment (plorenz) [reply router/config.go:687]: Yes, examples are in ziti

**Pull request merged**